test/test_metalearning/pyMetaLearn/test_meta_features.py

```diff
@@ -153,464 +152,598 @@ def meta_train_data_transformed(request):
+    X, y = multilabel_train_data
+    mf = meta_features.metafeatures["ClassEntropy"](X, y, logging.getLogger('Meta'))
+
+    classes = [(16, 84), (8, 92), (68, 32), (15, 85), (28, 72)]
+    entropies = []
+    for cls in classes:
+        cls = np.array(cls, dtype=np.float32)
+        cls = cls / sum(cls)
+        entropy = -np.sum([c * np.log2(c) for c in cls])
+        entropies.append(entropy)
+
+    assert pytest.approx(mf.value) == np.mean(entropies)
+
+
+def test_landmark_lda_multilabel(multilabel_train_data):
+    X, y = multilabel_train_data
+    mf = meta_features.metafeatures["LandmarkLDA"](X, y, logging.getLogger('Meta'))
+    assert np.isfinite(mf.value)
+
+
+def test_landmark_naive_bayes_multilabel(multilabel_train_data):
+    X, y = multilabel_train_data
+    mf = meta_features.metafeatures["LandmarkNaiveBayes"](X, y, logging.getLogger('Meta'))
+    assert np.isfinite(mf.value)
+
+
+def test_landmark_decision_tree_multilabel(multilabel_train_data):
+    X, y = multilabel_train_data
+    mf = meta_features.metafeatures["LandmarkDecisionTree"](X, y, logging.getLogger('Meta'))
+    assert np.isfinite(mf.value)
+
+
+def test_landmark_decision_node_multilabel(multilabel_train_data):
+    X, y = multilabel_train_data
+    mf = meta_features.metafeatures["LandmarkDecisionNodeLearner"](
+        X, y, logging.getLogger('Meta'))
+    assert np.isfinite(mf.value)
+
+
+def test_landmark_random_node_multilabel(multilabel_train_data):
+    X, y = multilabel_train_data
+    mf = meta_features.metafeatures["LandmarkRandomNodeLearner"](
+        X, y, logging.getLogger('Meta'))
+    assert np.isfinite(mf.value)
+
+
+def test_1NN_multilabel(multilabel_train_data):
+    X, y = multilabel_train_data
+    mf = meta_features.metafeatures["Landmark1NN"](X, y, logging.getLogger('TestMeta'))
+    assert np.isfinite(mf.value)
+
+
+def test_calculate_all_metafeatures_multilabel(multilabel_train_data):
+    meta_features.helper_functions.clear()
+    X, y = multilabel_train_data
+    categorical = {i: False for i in range(10)}
+    mf = meta_features.calculate_all_metafeatures(
+        X, y,  categorical, "Generated", logger=logging.getLogger('TestMeta'))
+    assert 52 == len(mf.metafeature_values)
+    sio = StringIO()
+    mf.dump(sio)
+
+
+def test_calculate_all_metafeatures_same_results_across_datatypes():
+    """
+    This test makes sure that numpy and pandas produce the same metafeatures.
+    This also is an excuse to fully test anneal dataset, and make sure
+    all metafeatures work in this complex dataset
+    """
+    X, y = fetch_openml(data_id=2, return_X_y=True, as_frame=True)
+    categorical = {col: True if X[col].dtype.name == 'category' else False
+                   for col in X.columns}
+    mf = meta_features.calculate_all_metafeatures(
+        X, y, categorical, "2", logger=logging.getLogger('Meta'))
+    assert 52 == len(mf.metafeature_values)
+    expected = {
+        # TODO: sync-up with matthias on landmarking
```

Comment:
I guess having accuracies match only for 5 digits or so is fine.

Reply:
The order of the columns depends on the format. Pandas do not reorder the columns after the column transformer, but numpy does.

The metafeatures that are dependent for the column order will produce a different result from pandas and for numpy. I added a comment on the code for this.